examples/filters/pycolorfilter/color_filter.py

```diff
@@ -0,0 +1,77 @@
+from typing import ClassVar, Mapping, Sequence, Optional, cast, Tuple, List, Any, Dict
+
+from typing_extensions import Self
+
+from viam.module.types import Reconfigurable
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName, ResponseMetadata, Geometry
+from viam.components.camera import Camera
+from viam.resource.types import Model, ModelFamily
+from viam.resource.base import ResourceBase
+from viam.media.video import NamedImage
+from PIL import Image
+from viam.errors import NoCaptureToStoreError
+from viam.services.vision import Vision
+
+class ColorFilterCam(Camera, Reconfigurable):
+    # A ColorFilterCam wraps the underlying camera `actual_cam` and only keeps the data captured on the actual camera if `vision_service`
+    # detects a certain color in the captured image.
+    MODEL: ClassVar[Model] = Model(ModelFamily("example", "camera"), "colorfilter")
+
+    def __init__(self, name: str):
+        super().__init__(name)
+
+    # Constructor
+    @classmethod
+    def new_cam(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        cam = cls(config.name)
+        cam.reconfigure(config, dependencies)
+        return cam
+
+    # Validates JSON Configuration
+    @classmethod
+    def validate_config(cls, config: ComponentConfig) -> Sequence[str]:
+        actual_cam = config.attributes.fields["actual_cam"].string_value
+        if actual_cam == "":
+            raise Exception("actual_cam attribute is required for a ColorFilterCam component")
+        vision_service = config.attributes.fields["vision_service"].string_value
+        if vision_service == "":
+            raise Exception("vision_service attribute is required for a ColorFilterCam component")
+        return [actual_cam, vision_service]
+
+    # Handles attribute reconfiguration
+    def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
+        actual_cam_name = config.attributes.fields["actual_cam"].string_value
+        actual_cam = dependencies[Camera.get_resource_name(actual_cam_name)]
+        self.actual_cam = cast(Camera, actual_cam)
+
+        vision_service_name = config.attributes.fields["vision_service"].string_value
+        vision_service = dependencies[Vision.get_resource_name(vision_service_name)]
+        self.vision_service = cast(Vision, vision_service)
+
+    # Returns details about the camera
+    async def get_properties(self, *, timeout: Optional[float] = None, **kwargs) -> Camera.Properties:
+        return await self.actual_cam.get_properties()
+
+    # Filters the output of the underlying camera
+    async def get_image(self, mime_type: str = "", *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs) -> Image.Image:
+        if extra and extra["fromDataManagement"]:
```

Comment:
Wondering if we want to have this `fromDataManagement` string as an importable variable in the SDKs as well? The only reason why I'm hesitating is because I don't want to add too many variables in the SDK that they then need to keep track of across languages, and because it would be easy for a user to debug any issues here (in contrast to NoCaptureToStoreError which is checked in the server code).

Reply:
I don't have a ton of context here; are you asking whether this "'special" variable `fromDataManagement` that's currently just encoded in the `extra` field should be a full parameter for `get_image`? I suppose I would prefer an actual parameter (which would require an API change if I'm not mistaken), but if this was purely an internally-specified variable only used by the data management team than I could maybe see the justification for leaving it like this. If a user ever needs to specify `fromDataManagement`, then a full parameter seems pretty necessary IMO.

Reply:
Not a full parameter (which would indeed be an API change and I believe a lot of extra work on the sdk side since fromDM is currently designed to be passed through the extra map for all collectors), but rather I was thinking of a variable that users could import - something like what we can do with the [go equivalent](https://github.com/viamrobotics/rdk/blob/ddd4ef604885ea94a45b64f5d5df5c68cc884c6e/examples/filters/colorfilter/color_filter.go#L139).

Reply:
flyby but I think adding something like `from_dm_from_extra` in a utils files might be valuable, like how we have `opid_from_metadata` in the python sdk. (in the sense that it's a bit manual and easy to get wrong but adding the helper reassures users that it's not that sketch and is the `suggested` way of accessing the var)

Reply:
Gotcha. If there's precedence in the Go module, I would also use a constant here. I see that Go uses a context key, but adding to the extra param seems ok for Python.

Reply:
Cool, thanks for the suggestions! I'll make a quick PR for this in the python SDK. And whoops sorry @benjirewis - I linked the wrong code pointer to my prev comment (fixed now)! The go camera is the only instance that needs to use the context key; the others should use `extra[data.FromDMString]`